infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/devfile/DockerimageComponentToWorkspaceApplier.java

```diff
@@ -257,6 +257,14 @@ static String toMachineName(String imageName) throws DevfileException {
       return imageName;
     }
 
+    if (imageName.length() > Names.MAX_CONTAINER_NAME_LENGTH) {
```

Comment:
Isn't it something that can be checked by JSON Schema?

Reply:
Well, we can't really restrict the names of the images. What would have to happen is conditional check that would mark alias required if the length of the image name would be > 63. Maybe it is doable in the schema but we have to have the check in the code anyway. So for simplicity's sake I only did it in the code. I believe the PR is writable by contributors so please add the schema changes if required (I don't have access to my computer this week).